presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -118,28 +119,32 @@
     private final ThriftHiveMetastoreStats stats;
     private final HiveCluster clientProvider;
     private final Function<Exception, Exception> exceptionMapper;
+    private final ColumnConverter columnConverter;
     private final boolean impersonationEnabled;
 
     @Inject
-    public ThriftHiveMetastore(HiveCluster hiveCluster, MetastoreClientConfig config)
+    public ThriftHiveMetastore(HiveCluster hiveCluster, MetastoreClientConfig config, ColumnConverter columnConverter)
     {
         this(
                 hiveCluster,
                 new ThriftHiveMetastoreStats(),
                 identity(),
+                columnConverter,
                 requireNonNull(config, "config is null").isMetastoreImpersonationEnabled());
     }
 
     public ThriftHiveMetastore(
             HiveCluster hiveCluster,
             ThriftHiveMetastoreStats stats,
             Function<Exception, Exception> exceptionMapper,
+            ColumnConverter columnConverter,
             boolean impersonationEnabled)
     {
         this.clientProvider = requireNonNull(hiveCluster, "hiveCluster is null");
         this.stats = requireNonNull(stats, "stats is null");
         this.exceptionMapper = requireNonNull(exceptionMapper, "exceptionMapper is null");
         this.impersonationEnabled = impersonationEnabled;
+        this.columnConverter = requireNonNull(columnConverter, "columnConverter is null");
```

Comment:
swap these two. The declaration of variables on stack, the parameters of the constructor, and the assignments within the constructor should have the same order.